### PR TITLE
Add callableWithURL

### PR DESCRIPTION
**Pull request by inlined**
Sorry for a late-coming urgent change; AFAICT nobody on the SDK team could staff these changes, so I'm trying to write the change for the top 3 SDKs before I/O freeze.

For context, see [go/cf3v2-crash-landing](http://go/cf3v2-crash-landing).
For API approval, see [go/callable-functions-custom-urls](http://go/callable-functions-custom-urls)

### CODE REVIEW

File: FirebaseFunctions/Sources/Callable+Codable.swift
Comment (paulb777) [inline FirebaseFunctions/Sources/Callable+Codable.swift:19]: Please update the PR to be just the new API implementation and I'll take another look

**Comment by inlined** [comment]
Jebus. Sorry about that. I couldn't get ./scripts/check to work at first so I just ran `clag-formatter` and blindly pushed the results. That was crazy. Dropped the commits and rerun after I got `scripts/check.sh` to work.

File: FirebaseFunctions/Sources/HTTPSCallable.swift
Comment (paulb777) [inline FirebaseFunctions/Sources/HTTPSCallable.swift:103]: Don't use force unwraps. Instead do `if let name = name` {`

File: FirebaseFunctions/Sources/Functions.swift
Comment (paulb777) [inline FirebaseFunctions/Sources/Functions.swift:107]: It could be confusing to have a String parameter named url. Any reason for the API not taking a URL parameter (NSURL in Objective C)?

**Review by paulb777 — CHANGES_REQUESTED** [review]
Thanks for the updates.

Please add tests to:

- Tests/Unit/FunctionsTests.swift
- Tests/Unit/FunctionsAPITests.swift
- Tests/Integration/IntegrationTests.swift
- Tests/ObjCIntegration/FIRIntegrationTests.m
- Tests/ObjCIntegration/ObjCAPITests.m
- Tests/ObjCIntegration/ObjCPPAPITests.mm

If necessary, update FirebaseFunctions/README.md and the scripts it uses for running the integration tests on the emulator.

Organize a post-mortem so we can address the factors that led to this PR coming so close to code freeze.

File: FirebaseFunctions/Sources/HTTPSCallable.swift
Comment (paulb777) [inline FirebaseFunctions/Sources/HTTPSCallable.swift:108]: `else if let url = url`

I don't have a strong opinion about generating an error in a final `else` or ignoring it.

**Review by paulb777 — APPROVED** [review]
LGTM, after one more nit. Thanks for working through the comments.

Please give @ryanwilson a chance to review tomorrow morning before merging.

File: FirebaseFunctions/Tests/ObjCIntegration/ObjCAPITests.m
Comment (ryanwilson) [inline FirebaseFunctions/Tests/ObjCIntegration/ObjCAPITests.m:36]: Is there still a test that uses the old `name`?

File: FirebaseFunctions/Tests/ObjCIntegration/ObjCPPAPITests.mm
Comment (ryanwilson) [inline FirebaseFunctions/Tests/ObjCIntegration/ObjCPPAPITests.mm:38]: Same as above

File: FirebaseFunctions/Tests/Unit/FunctionsAPITests.swift
Comment (ryanwilson) [inline FirebaseFunctions/Tests/Unit/FunctionsAPITests.swift:53]: Can you also add one for the Codable API? I realize now that we don't have it for the other methods so let me know if you need a hand generating it.

**Review by ryanwilson — COMMENTED** [review]
LGTM with some minor nits - thanks!

File: FirebaseFunctions/Tests/ObjCIntegration/ObjCAPITests.m
Comment (ryanwilson) [reply FirebaseFunctions/Tests/ObjCIntegration/ObjCAPITests.m:36]: Resolved - didn't realize these were just the API tests.

File: FirebaseFunctions/Tests/ObjCIntegration/ObjCPPAPITests.mm
Comment (ryanwilson) [reply FirebaseFunctions/Tests/ObjCIntegration/ObjCPPAPITests.mm:38]: Resolved, didn't realize they were just API tests.

File: FirebaseFunctions/Tests/Unit/FunctionsAPITests.swift
Comment (inlined) [reply FirebaseFunctions/Tests/Unit/FunctionsAPITests.swift:54]: Figured it out.

Comment (ryanwilson) [reply FirebaseFunctions/Tests/Unit/FunctionsAPITests.swift:54]: That isn't the API surface that folks use and `new` isn't a keyword in Swift so I'm not sure how it's passing if it is - I have a change locally that adds the full tests just building locally to verify

**Pull request merged**
